spanner/errors.go

```diff
@@ -20,12 +20,18 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/googleapis/gax-go/v2/apierror"
+
 	"google.golang.org/genproto/googleapis/rpc/errdetails"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
 // Error is the structured error returned by Cloud Spanner client.
+//
+// Deprecated: The APIError should be extracted from the wrapped error by
```

Comment:
I think this deprecation warning could use more elaborate instructions on exactly what users should do. The end goal is that the client should return an `APIError`, and we want to instruct the users that they should also use `APIError` to get the details from the error that is returned by the client. The safest way that users can do that (at least that I can think of) would be the following:

```
	// Unwrap any error that is returned by the Spanner client as an APIError
	// to access the error details. Do not try to convert the error to the
	// spanner.Error struct, as that struct may be removed in a future release.
	//
	// Example:
	// var apiErr *apierror.APIError
	// _, err := spanner.NewClient(context.Background())
	// errors.As(err, &apiErr)
```

This example code will work both when the client returns instances of `spanner.Error` and when it starts to return `apierror.APIError` instances.

It might also be good to add a test to the client that verifies the above, so that we don't accidentally break this in the future.